File: Leetcode/invertTree.java
public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        
        TreeNode temp = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(temp);
        return root;
    }


public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        List<TreeNode> queue = new ArrayList<TreeNode>();
        queue.add(root);
        while (!queue.isEmpty()) {
            TreeNode node = queue.remove(0);
            TreeNode left = node.left;
            node.left = node.right;
            node.right = left;
            if (node.left != null) {
                queue.add(node.left);
            }
            if (node.right != null) {
                queue.add(node.right);
            }
        }
        return root;
    }

File: Leetcode/invertTree.swift
/*
func invertTree(root: TreeNode?) -> TreeNode? {
if root == nil {
return nil
}
var temp = root!.left
root!.left = invertTree(root!.right)
root!.right = invertTree(temp)
return root
}
*/
func invertTree(root: TreeNode?) -> TreeNode? {
    if root == nil {
        return nil
    }
    
    var queue: [TreeNode] = []
    queue.append(root!)
    while !queue.isEmpty {
        var node: TreeNode = queue.removeFirst()
        var left = node.left
        node.left = node.right
        node.right = left
        
        if node.right != nil {
            queue.append(node.right!)
        }
        if node.left != nil {
            queue.append(node.left!)
        }
    }
    return root
}
